cmd/ax/query.go

```diff
@@ -80,6 +91,74 @@ func buildFilters(wheres []string) []common.QueryFilter {
 	return filters
 }
 
+var membershipFilterRegex = regexp.MustCompile(`([^!=<>]+)\s*:\s*(.*)`)
+
+func buildMembershipFilters(oneOfs []string, notOneOfs []string) []common.MembershipFilter {
+	// Build a nested map of field names and their membership constraints
+	fields := make(map[string]map[bool][]string)
```

Comment:
Wouldn't a `map[string]struct{ oneOf bool, value string}` be nicer? This is what this encodes right?

Reply:
Maybe, but this way we can just drop in the lists we generate.